### PR TITLE
Create initializeAggregation to initialize an aggregation function

### PR DESCRIPTION
**Pull request by YiuRULE**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add a function initializedAggregation to initialize an aggregation based on a single value

Related to https://github.com/ClickHouse/ClickHouse/issues/11866

### CODE REVIEW

File: src/Functions/initializeAggregation.cpp [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+            that = func->getNestedFunction().get();
+        that->addBatch(input_rows_count, places.data(), 0, aggregate_arguments, arena.get());
+    }
+
+    for (size_t i = 0; i < input_rows_count; ++i)
+        if (!res_col_aggregate_function)
+            agg_func.insertResultInto(places[i], res_col, arena.get());
+        else
+            res_col_aggregate_function->insertFrom(places[i]);
+    block.getByPosition(result).column = std::move(result_holder);
```
Comment (CurtizJ) [inline src/Functions/initializeAggregation.cpp:152]: If it's the exact code from `arrayReduce` function, maybe move it to seperate funtion with name like `applyAggregation`.

File: src/Functions/initializeAggregation.cpp [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    const ColumnConst * aggregate_function_name_column = checkAndGetColumnConst<ColumnString>(arguments[0].column.get());
+    if (!aggregate_function_name_column)
+        throw Exception("First argument for function " + getName() + " must be constant string: name of aggregate function.",
+            ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+    DataTypes argument_types(arguments.size() - 1);
+    for (size_t i = 1, size = arguments.size(); i < size; ++i)
+    {
+        argument_types[i - 1] = arguments[i].type;
+    }
```
Comment (CurtizJ) [inline src/Functions/initializeAggregation.cpp:66]: A shorter way to do it `DataTypes argument_types(arguments.begin() + 1, arguments.end())`

File: src/Functions/initializeAggregation.cpp (hunk 1 above)
Comment (YiuRULE) [reply src/Functions/initializeAggregation.cpp:152]: not the exact same code, we accept different kind of arguments and we have different function call to execute the aggregation, does it's worth it ?

Comment (CurtizJ) [reply src/Functions/initializeAggregation.cpp:152]: I see, `addBatch` vs `addBatchArray`. Ok, let leave it as is.

**Review by CurtizJ — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
@CurtizJ Need to check if it works for functions with zero arguments (`count`).

**Comment by alexey-milovidov** [comment]
Also what shall we do with a function to initialize empty aggregation state with given types of arguments?
If we will have a sepate function for that purpose, what will be the name of this function?

**Comment by AnaUvarova** [comment]
Hi, @CurtizJ, 
hit me with a hint please: 
what exact types will this function return as a result of aggregating my input lines?

**Comment by CurtizJ** [comment]
The return type will be the same as the return type of function, that `initializeAgregation` takes as first argument. In case of functions with suffix `State`, for which `initializeAgregation` is intended for, the return type will be [`AggregateFunction`](https://clickhouse.tech/docs/en/sql-reference/data-types/aggregatefunction/).

**Comment by AnaUvarova** [comment]
> The return type will be the same as the return type of function, that `initializeAgregation` takes as first argument. In case of functions with suffix `State`, for which `initializeAgregation` is intended for, the return type will be [`AggregateFunction`](https://clickhouse.tech/docs/en/sql-reference/data-types/aggregatefunction/).

Thank you.

**Pull request merged**